src/smc-webapp/project/settings/body.tsx

```diff
@@ -0,0 +1,253 @@
+}
+
+interface ReduxProps {
+  get_total_upgrades: Function;
+  groups: string[];
+
+  kucalc: string;
+
+  get_course_info: Function;
+  get_total_upgrades_you_have_applied: Function;
+  get_upgrades_you_applied_to_project: Function;
+  get_total_project_quotas: Function;
+  get_upgrades_to_project: Function;
+  compute_images: Map<string, any>;
+  all_projects_have_been_loaded: boolean;
+
+  configuration: Map<string, any>;
+  available_features: object;
+}
+
+export const Body = rclass<ReactProps>(
+  class Body extends React.Component<ReactProps & ReduxProps> {
+    public static reduxProps({ name }) {
+      return {
+        account: {
+          get_total_upgrades: rtypes.func,
+          groups: rtypes.array
+        },
+        customize: {
+          kucalc: rtypes.string
+        },
+        projects: {
+          get_course_info: rtypes.func,
+          get_total_upgrades_you_have_applied: rtypes.func,
+          get_upgrades_you_applied_to_project: rtypes.func,
+          get_total_project_quotas: rtypes.func,
+          get_upgrades_to_project: rtypes.func,
+          compute_images: rtypes.immutable.Map,
+          all_projects_have_been_loaded: rtypes.bool
+        },
+        [name]: {
+          configuration: rtypes.immutable,
+          available_features: rtypes.object
+        }
+      };
+    }
+
+    shouldComponentUpdate(props) {
+      return (
+        misc.is_different(this.props, props, [
+          "project",
+          "user_map",
+          "project_map",
+          "compute_images",
+          "configuration",
+          "available_features",
+          "all_projects_have_been_loaded"
+        ]) ||
+        (props.customer != undefined &&
+          !props.customer.equals(this.props.customer))
+      );
+    }
+
+    render() {
+      // get the description of the share, in case the project is being shared
+      const id = this.props.project_id;
+
+      const upgrades_you_can_use = this.props.get_total_upgrades();
+
+      const course_info = this.props.get_course_info(this.props.project_id);
+      const upgrades_you_applied_to_all_projects = this.props.get_total_upgrades_you_have_applied();
+      const upgrades_you_applied_to_this_project = this.props.get_upgrades_you_applied_to_project(
+        id
+      );
+      const total_project_quotas = this.props.get_total_project_quotas(id); // only available for non-admin for now.
+      const all_upgrades_to_this_project = this.props.get_upgrades_to_project(
+        id
+      );
+      const allow_urls = redux
+        .getStore("projects")
+        .allow_urls_in_emails(this.props.project_id);
+
+      const { commercial } = require("../../customize");
+
+      const { is_available } = require("../../project_configuration");
+      const available = is_available(this.props.configuration);
+      const have_jupyter_lab = available.jupyter_lab;
+      const have_jupyter_notebook = available.jupyter_notebook;
+      console.log("name?", this.props.name)
```

Comment:
I assume this shouldn't be here.   Can you somehow check your PR (from before it to HEAD) for console.logs and delete or comment them out?

Reply:
(i just deleted the console.log I saw in a commit that I pushed)

Reply:
this is the only one.